### ISSUE TITLE
Docker minimum Triton example seems incomplete

### ISSUE THREAD

**Issue by bryanmcgrane** [triton-inference-server/server#2910]
**Description**
I'm trying to get a minimum deployment working with Triton, and am following the doc here: https://github.com/triton-inference-server/server/blob/main/docs/compose.md

I can't seem to get things working.

**Triton Information**
I'm using version 21.02

**Are you using the Triton container or did you build it yourself?**
I am trying to build it myself.

**To Reproduce**
Below is my Dockerfile

```
FROM nvcr.io/nvidia/tritonserver:21.02-py3 as full
FROM nvcr.io/nvidia/tritonserver:21.02-py3-min

EXPOSE 8001/tcp

COPY --from=full /opt/tritonserver/bin /opt/tritonserver/bin
COPY --from=full /opt/tritonserver/lib /opt/tritonserver/lib
COPY --from=full /opt/tritonserver/backends/tensorflow2 /opt/tritonserver/backends/tensorflow2

CMD ["opt/tritonserver/bin/tritonserver"]
```

When I try to run the Dockerfile I see the following error:
`opt/tritonserver/bin/tritonserver: error while loading shared libraries: libre2.so.5: cannot open shared object file: No such file or directory`

**Expected behavior**
Triton server runs in a minimal configuration.

**Comment by CoderHam** [comment]
You will likely need to install libre2 and a few other deps that we need for building and running Triton in the -min container.
https://github.com/triton-inference-server/server/blob/r21.05/build.py#L490
I believe the min image does not contain these libraries and we will need to update the docs accordingly. Thanks for bringing this to our attention.

**Comment by bryanmcgrane** [comment]
After installing libre2-dev and libb64-dev I ran into this error:

```
./opt/tritonserver/bin/tritonserver: error while loading shared libraries: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
```

**Comment by bryanmcgrane** [comment]
And after installing all the packages in your link I hit the same error as well.

**Comment by CoderHam** [comment]
NVML and therefore GPU metrics are not supported on Jetson. You will have to build Triton with GPU Metrics disabled for Jetson

**Comment by bryanmcgrane** [comment]
Got it. I didn't build Triton, necessarily. I just copied the binary from the non-minified triton docker image. How can I disable GPU metrics? Is there a doc for that?

**Comment by CoderHam** [comment]
@bryanmcgrane not sure that is wise since the Docker Triton was built for x86 and not Aarch64. I'd just use the tar file for Jetson in our release notes. 

If you still want to proceed with your path you should refer to the build docs. There is a build.py script which is the best way to build Triton. Again I do not recommend copying binaries and libraries built for x86 in the Docker and using them on Jetson.

**Comment by bryanmcgrane** [comment]
I might be missing something. If I build and run the following Docker image everything works just fine:

```
FROM nvcr.io/nvidia/tritonserver:21.02-py3
CMD ["tritonserver"]
```

But this Docker image has the missing `libnvidia-ml.so.1` lib. So it seems like there is a larger difference between the full image and the minified image. 

```
FROM nvcr.io/nvidia/tritonserver:21.02-py3 as full
FROM nvcr.io/nvidia/tritonserver:21.02-py3-min

COPY --from=full /opt/tritonserver/bin /opt/tritonserver/bin
COPY --from=full /opt/tritonserver/lib /opt/tritonserver/lib
COPY --from=full /opt/tritonserver/backends/tensorflow2 /opt/tritonserver/backends/tensorflow2

CMD ["/opt/tritonserver/bin/tritonserver"]
```

**Comment by CoderHam** [comment]
Please ignore my comments regarding Jetson. I misread and thought you were building for Jetson. NVML is supported and you should not have to make any changes to the build. 
So yes, your conclusion is correct. The min image is missing  libnvidia-ml.so.1 as well. You will likely have to install this in the min image as well. 
We will work on fixing the docs to avoid the confusion.

**Comment by bryanmcgrane** [comment]
That worked! Here's my working dockerfile with a tensorflow2 backend

```
FROM nvcr.io/nvidia/tritonserver:21.02-py3 as full
FROM nvcr.io/nvidia/tritonserver:21.02-py3-min

RUN apt-get update && \
    apt-get install -y --no-install-recommends \
    libnvidia-ml-dev \
    libre2-dev \
    libb64-dev

COPY --from=full /opt/tritonserver/bin /opt/tritonserver/bin
COPY --from=full /opt/tritonserver/lib /opt/tritonserver/lib
COPY --from=full /opt/tritonserver/backends/tensorflow2 /opt/tritonserver/backends/tensorflow2
```

**Comment by CoderHam** [comment]
@bryanmcgrane thanks for sharing the same. We will fix the instructions for the min container accordingly.

**Comment by deadeyegoodwin** [comment]
This will be fixed by #3039 .

**Comment by tanmayv25** [comment]
Closing as #3039 was merged.
